stock_picking_mail_incorrect_qty/models/stock_picking.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2021 Coop IT Easy SCRL fs
+#   Carmen Bianca Bakker <carmen@coopiteasy.be>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class Picking(models.Model):
+    _inherit = "stock.picking"
+
+    zero_received_move_ids = fields.One2many(
+        "stock.move",
+        string="Stock moves that were not received",
+        compute="_compute_zero_received",
+    )
+    too_few_received_move_ids = fields.One2many(
+        "stock.move",
+        string="Stock moves of which too few were received",
+        compute="_compute_too_few_received",
+    )
+    too_many_received_move_ids = fields.One2many(
+        "stock.move",
+        string="Stock moves of which too many were received",
+        compute="_compute_too_many_received",
+    )
+
+    form_view_url = fields.Char(
+        string="Form View URL", compute="_compute_form_view_url"
+    )
+
+    @api.multi
+    @api.depends("move_lines")
+    def _compute_zero_received(self):
+        for pick in self:
+            pick.zero_received_move_ids = pick._filtered_moves().filtered(
+                lambda move: move.quantity_done == 0 and move.product_qty
+            )
+
+    @api.multi
+    @api.depends("move_lines")
+    def _compute_too_few_received(self):
+        for pick in self:
+            pick.too_few_received_move_ids = pick._filtered_moves().filtered(
+                lambda move: move.quantity_done < move.product_qty
+            )
+
+    @api.multi
+    @api.depends("move_lines")
+    def _compute_too_many_received(self):
+        for pick in self:
+            pick.too_many_received_move_ids = pick._filtered_moves().filtered(
+                lambda move: move.quantity_done > move.product_qty
+            )
+
+    @api.multi
+    def _compute_form_view_url(self):
+        for pick in self:
+            pick.form_view_url = (
+                "{}/web#id={}&model=stock.picking&view_type=form".format(
+                    pick.env["ir.config_parameter"].sudo().get_param("web.base.url"),
+                    pick.id,
+                )
+            )
```

Comment:
I think it's better to call the action opening the picking form view.

Reply:
I was going to ask about this. How do you call that action? Furthermore, is a computed field the correct place to do this? I need access to the URL in the template, and I don't have access to a lot of stuff inside of the template.

Is it possible to pass additional context into the template, so that I don't need to rely on a computed field at all?

Reply:
Mmmh, if it's the email, calling the action won't work :/ So building the url manually is probably the best way to go. But you don't have to use a computed field, you can call a method from the object passed to the template.

Reply:
Moved this into a function called `form_view_url()`. It has to be public for the mail template.